src/MongoDB.Driver/Linq/Translators/PredicateTranslator.cs

```diff
@@ -952,15 +964,17 @@ private FilterDefinition<BsonDocument> TranslatePipelineAny(PipelineExpression n
             FilterDefinition<BsonDocument> filter;
             var renderWithoutElemMatch = CanAnyBeRenderedWithoutElemMatch(whereExpression.Predicate);
 
+            var fieldName = PrepareFieldName(fieldExpression);
             if (renderWithoutElemMatch)
             {
-                var predicate = FieldNamePrefixer.Prefix(whereExpression.Predicate, fieldExpression.FieldName);
+                var predicate = FieldNamePrefixer.Prefix(whereExpression.Predicate, fieldName);
```

Comment:
The changing fieldName for this code path is checked in this test: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/53/files#diff-28fd5e9de3efd662172dc9b076ed2c10R290